ordering/__init__.py

```diff
@@ -114,8 +123,8 @@ def __init__(self, ordering: Ordering[T], item: T) -> None:
 
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, OrderingItem):
-            return bool(self.item == other)
```

Comment:
Not for me it doesn't, using the `--strict` option all I get is:
```
ordering\__init__.py:74: warning: Returning Any from function with declared return type "builtins.bool"
```
which pertains to returning `NotImplemented` from `Ordering.__eq__`.

I can partially understand why this would be the case this, as there is no actual requirement for `__eq__` (or any comparison method for that matter) to actually return a `bool`, just to return an object with a `__bool__` method, but since `object` has a `__bool__` method, that's every object. Thus I'd say the more technically correct option would be to change the return type to `object`, however mypy considers that to be an error too.

It's also odd that this line gives you a typing error when the one below doesn't since the typing of the `and` operator is `((a: A) and (b: B)): Union[A, B]`. So I'm thinking this could be a error with mypy?